plugins/hydra_submitit_launcher/hydra_plugins/hydra_submitit_launcher/submitit_launcher.py

```diff
@@ -96,36 +100,40 @@ def launch(
 
         num_jobs = len(job_overrides)
         assert num_jobs > 0
-
-        # make sure you don't change inplace
-        queue_parameters = self.queue_parameters.copy()
-        OmegaConf.set_struct(queue_parameters, True)
-        executors = {
-            QueueType.auto: submitit.AutoExecutor,
-            QueueType.slurm: submitit.SlurmExecutor,
-            QueueType.local: submitit.LocalExecutor,
+        exec_name = self.params.executor.value
+        params = self.params
+
+        # build executor
+        init_renamer = dict(executor="cluster", submitit_folder="folder")
+        init_params = {name: params[k] for k, name in init_renamer.items()}
+        init_params = {
+            k: v.value if isinstance(v, Enum) else v for k, v in init_params.items()
         }
-        init_parameters = {"cluster", "max_num_timeout", "slurm_max_num_timeout"}
-        executor = executors[self.queue](
-            folder=self.folder,
+        specific_init_keys = {"max_num_timeout"}
+        init_params.update(
             **{
-                x: y
-                for x, y in queue_parameters[self.queue.value].items()
-                if x in init_parameters
-            },
-        )
-        executor.update_parameters(
-            **{
-                x: y
-                for x, y in queue_parameters[self.queue.value].items()
-                if x not in init_parameters
+                f"{exec_name}_{x}": y
+                for x, y in params.items()
+                if x in specific_init_keys
             }
         )
+        init_keys = specific_init_keys | set(init_renamer)  # used config keys
+        executor = submitit.AutoExecutor(**init_params)
```

Comment:
Now everything goes through AutoExecutor, whatever the choice of the executor. This makes it more maintainable.

Reply:
As you can tell, it takes some trickery to prepare the correct dictionary for the AutoExecutor.
The reason is that the input config is no longer lining up with AutoExecutor parameters.

If you used SlurmExecutor and LocalExecutor directly, with corresponding SubmititSlurmLauncher and SubmititLocalLauncher as I suggested above I think you would be able to more easily match the init signature of each to what you want to pass the respective submitit launcher.

You can still use auto, but at least in this case the logic in each of the launchers will need to initialize a known mode which will simplify things

Reply:
> If you used SlurmExecutor and LocalExecutor directly, with corresponding SubmititSlurmLauncher and SubmititLocalLauncher as I suggested above I think you would be able to more easily match the init signature of each to what you want to pass the respective submitit launcher.

I *won't* call `SlurmExecutor` and `LocalExecutor` directly, that would make the plugin harder to maintain. The alignment will actually be much simpler to maintain. Also, there's a PR on its way in submitit that would make it even easier.